db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementPermission.java

```diff
@@ -159,6 +159,10 @@ public String getObjectType()
 					Authorizer.PUBLIC_AUTHORIZATION_ID,
 					dbo);
 			}
+			if (rd == null && lcc.getCurrentGroupUser(activation) != null) {
```

Comment:
Similar to StatementSchemaPermission.java, we need to add check 
		if( !isCorrectPermission( perm ) ) { perm = getPermissionDescriptor(lcc.getCurrentGroupUser(activation), dd ); }